src/main/scala/GithubRelease.scala

```diff
@@ -47,33 +48,43 @@ case object GithubRelease {
       typeMap.getContentType
     }
 
-    def ghreleaseGetCredentials: DefTask[GitHub] = Def.task {
-      val log = streams.value.log
-      val conf = file(System.getProperty("user.home")) / ".github"
-
-      while (!conf.exists || !GitHub.connect.isCredentialValid) {
-        log.warn("Your github credentials for sbt-github-release plugin are not set yet")
-        SimpleReader.readLine("Go to https://github.com/settings/applications \ncreate a new token and paste it here: ") match {
-          case Some(token) => {
-            try {
-              val gh = GitHub.connectUsingOAuth(token)
-              if (gh.isCredentialValid) {
-                IO.writeLines(conf, Seq("oauth = " + token))//, append = true)
-                log.info("Wrote OAuth token to " + conf)
-              }
-            } catch {
-              case e: Exception => log.error(e.toString)
-            }
-          }
-          case _ => sys.error("If you want to use sbt-github-release plugin, you should set credentials correctly")
-        }
+    private val defaultPropertiesFile: sbt.File = file(s"${getProperty("user.home")}/.github")
+
+    def readFromEnvironment(): Option[String] = {
+      readFromEnvironment("GITHUB_TOKEN")
+    }
+
+    def readFromEnvironment(environmentVariableName: String): Option[String] = {
+      System.getenv().asScala.get(environmentVariableName)
+    }
+
+    def readCredentialsFrom(): Option[String] = {
+      readCredentialsFrom(defaultPropertiesFile)
+    }
+
+    def readCredentialsFrom(file: File): Option[String] = {
```

Comment:
I suggest to rename this method to `ghreleaseTokenFromFile` or `ghreleaseGithubTokenFromFile`